### ISSUE TITLE
write/plot bug when root at leaf

### ISSUE THREAD

**Issue by crsl4** [crsl4/PhyloNetworks.jl#129]
I ran into an erratic behavior in which a leaf disappears in the parenthetical format, but it is there in the `HybridNetwork` object. See below.

Leaf "B" is missing from the parenthetical format:
```
julia> net1
HybridNetwork, Un-rooted Network
10 edges
10 nodes: 5 tips, 1 hybrid nodes, 4 internal tree nodes.
tip labels: B, E, C, D, ...
(((A:9.796,(D:9.769,(C:5.647,#H19:0.323::0.494):2.799):3.326):3.449,(E:0.0)#H19:0.323::0.506):9.748);
```

But "B" is there in the `HybridNetwork` object:
```
printNodes(net1)
node leaf  hybrid hasHybEdge name inCycle edges'numbers
-3   true  false  false      B    -1      3   
-4   true  false  false      E    -1      20  
-7   true  false  false      C    -1      22  
-8   true  false  false      D    -1      9   
-6   false false  true            19      6    3    23  
-9   true  false  false      A    -1      16  
16   false false  false           19      16   6    12  
18   false false  false           19      12   9    21  
19   false true   true       H19  19      20   24   23  
20   false false  true            19      21   22   24
```

I traced this error to the fact that the root is placed at leaf B, and if I change the root, everything is ok:
```
julia> net1.root
1 ##first node is B

julia> rootatnode!(net1,16)
HybridNetwork, Rooted Network
10 edges
10 nodes: 5 tips, 1 hybrid nodes, 4 internal tree nodes.
tip labels: B, E, C, D, ...
(A:9.796,(B:9.748,(E:0.0)#H19:0.323::0.506):3.449,(D:9.769,(C:5.647,#H19:0.323::0.494):2.799):3.326);
```
Note that not only `writeTopology` is affected by the root in the leaf, also `plot`.

This issue reminded me of an [old issue](https://github.com/crsl4/PhyloNetworks.jl/issues/84) which we now closed. At that time, @cecileane fixed `deleteleaf`, but I think that what we should fix is `writeTopology` and/or `plot` to make them robust to the root position.

What do you think?

**Comment by cecileane** [comment]
What happens if you use option `internallabel=true` in `writeTopology`? I suspect that `B` will come back, at the root (not as a leaf). I don't think it is a bug for `writeTopology`. The root should never be placed at a leaf ideally: the phylogeny would not an "X-tree" as Semple & Stell define it, I think, in that case. Many other problems could occur, when the root is placed at a leaf. Plots don't work, as a start.

As a start, `writeTopology` could check if the root is a leaf, send a warning, and write the name in that case.

**Comment by cecileane** [comment]
`writeTopology` uses `internallabel=true` by default. But `Base.show` of a network calls `writeTopology` with the option `internallabel=false` (and branch lengths rounded to 3 digits etc.).

Should `Base.show` instead use the option `internallabel=true`? Would would be satisfied with this fix?

Fixing the plots is going to be harder.

**Comment by crsl4** [comment]
First, yes, let's have `Base.show` have the `internallabel=true` by default. Although this option does not loot that pretty:
```
julia> writeTopology(net1)
"(((A:9.79598494514856,(D:9.768690388782732,(C:5.646708668821205,#H19:0.32315290025263166::0.49363924382404717):2.799203466372674):3.3261359319166255):3.4491084083233456,(E:0.0)#H19:0.3232194260400639::0.5063607561759529):9.747740390554483)B;"
```

Currently, `plot` does show a warning (and that's how I found out about the problem). So, maybe `plot` is ok as it is.

Maybe having `writeTopology` check if the root is in a leaf and throw a warning too would be good in this case.

The network here was produced by snaq (this happens because the `net1.root=1` attribute is never changed, but nodes are re-used and moved around, and in this case, the first node ended up being a leaf). So maybe adding a check at the end of snaq so that `net.root` is not pointing to a leaf.

**Comment by cecileane** [comment]
All good ideas. Will you work on it @crsl4, since the second aspect affects `snaq!`?

**Comment by crsl4** [comment]
Definitely! I was just waiting for your green light :)

**Comment by crsl4** [comment]
Fixed with commit [1da33251](https://github.com/crsl4/PhyloNetworks.jl/commit/1da33251d9c92dff82f9e994a98eee1a9004d646) by checking the root in the multiple alleles case. This is a band-aid fix for now as I think `snaq` and `optTopRuns` could be improved for the multiple alleles case, but will need more drastic changes.

Currently, in the multiple alleles case, we do:
- In `optTopRuns`:
    1. check the root
    2. print maxNet (with `writeTopologyLevel1` which merges leaves and checks root again inside)
- In `snaq`:
    1. merge leaves for good before returning the network
    2. checks the root again

A lot of repetition! In my to-do list to fix.
